gui/src/mainwindow.cpp

```diff
@@ -608,103 +612,101 @@ void MainWindow::receivedTcpMessage(TcpMessage tcpMessage)
         return;
-    }
-    if (msgID == TCP_MSG_KEY::MSG_UBX_CFG_TP5) {
-        UbxTimePulseStruct tp;
+    } else if (msgID == TCP_MSG_KEY::MSG_UBX_CFG_TP5) {
+        UbxTimePulseStruct tp {};
         *(tcpMessage.dStream) >> tp;
         emit gpsTP5Received(tp);
         return;
-    }
-    if (msgID == TCP_MSG_KEY::MSG_HISTOGRAM) {
-        Histogram h;
+    } else if (msgID == TCP_MSG_KEY::MSG_HISTOGRAM) {
+        Histogram h {};
         *(tcpMessage.dStream) >> h;
         emit histogramReceived(h);
         return;
-    }
-    if (msgID == TCP_MSG_KEY::MSG_ADC_MODE) {
+    } else if (msgID == TCP_MSG_KEY::MSG_ADC_MODE) {
         quint8 mode;
         *(tcpMessage.dStream) >> mode;
         emit adcModeReceived(mode);
         return;
-    }
-    if (msgID == TCP_MSG_KEY::MSG_LOG_INFO) {
+    } else if (msgID == TCP_MSG_KEY::MSG_LOG_INFO) {
         LogInfoStruct lis;
         *(tcpMessage.dStream) >> lis;
         emit logInfoReceived(lis);
         return;
-    }
-    if (msgID == TCP_MSG_KEY::MSG_UBX_TIMEMARK) {
+    } else if (msgID == TCP_MSG_KEY::MSG_UBX_TIMEMARK) {
         UbxTimeMarkStruct tm;
         *(tcpMessage.dStream) >> tm;
         emit timeMarkReceived(tm);
         return;
-    }
-    if (msgID == TCP_MSG_KEY::MSG_MQTT_STATUS) {
+    } else if (msgID == TCP_MSG_KEY::MSG_MQTT_STATUS) {
         bool connected = false;
         *(tcpMessage.dStream) >> connected;
-        emit mqttStatusChanged(connected);
+        if (tcpMessage.dStream->atEnd()) {
+            emit mqttStatusChanged(connected);
+        } else {
+            int extStatus { -1 };
+            *(tcpMessage.dStream) >> extStatus;
+            MuonPi::MqttHandler::Status status { extStatus };
+            emit mqttStatusChanged(status);
+        }
         return;
-    }
-    if (msgID == TCP_MSG_KEY::MSG_POLARITY_SWITCH) {
+    } else if (msgID == TCP_MSG_KEY::MSG_POLARITY_SWITCH) {
         bool pol1;
         bool pol2;
         *(tcpMessage.dStream) >> pol1 >> pol2;
         emit polaritySwitchReceived(pol1, pol2);
-        updateUiProperties();
         return;
-    }
-    if (msgID == TCP_MSG_KEY::MSG_GPIO_INHIBIT) {
+    } else if (msgID == TCP_MSG_KEY::MSG_GPIO_INHIBIT) {
         bool inhibit;
         *(tcpMessage.dStream) >> inhibit;
         emit gpioInhibitReceived(inhibit);
-        updateUiProperties();
         return;
-    }
-    if (msgID == TCP_MSG_KEY::MSG_MQTT_INHIBIT) {
+    } else if (msgID == TCP_MSG_KEY::MSG_MQTT_INHIBIT) {
         bool inhibit;
         *(tcpMessage.dStream) >> inhibit;
         emit mqttInhibitReceived(inhibit);
-        updateUiProperties();
+        return;
+    } else if (msgID == TCP_MSG_KEY::MSG_VERSION) {
+        MuonPi::Version::Version hw_ver, sw_ver;
+        *(tcpMessage.dStream) >> hw_ver >> sw_ver;
+        emit daemonVersionReceived(hw_ver, sw_ver);
+        return;
+    } else {
+        qDebug() << "received unknown TCP message, msgID =" << QString::number(static_cast<int>(msgID));
```

Comment:
this should rather be a warning, shouldn't it?
```suggestion
        qWarning() << "received unknown TCP message, msgID =" << QString::number(static_cast<int>(msgID));
```